pkg/registry/pod/strategy.go

```diff
@@ -153,6 +154,23 @@ func (podStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return validation.ValidatePodStatusUpdate(obj.(*api.Pod), old.(*api.Pod))
 }
 
+func CheckPod(r *registry.Store, ctx api.Context, err error, obj runtime.Object) error {
+	if !errors.IsAlreadyExists(err) {
+		return err
+	}
+	pod := obj.(*api.Pod)
+	retObj, getPodErr := r.Get(ctx, pod.Name)
+	if getPodErr != nil {
+		return err
+	}
+	retPod := retObj.(*api.Pod)
+	if api.IsPodTerminated(retPod) {
+		status_err, _ := err.(*errors.StatusError)
+		status_err.ErrStatus.Message += fmt.Sprintf("\n Existant pod is in terminated state! Please check its status and delete it!\n Use `kubectl get pods/%s` \n", pod.Name)
```

Comment:
This check is called in case of failure of CREATE obj operation with KeyAlreadyExists error: so there is no retrieved or created object and to check object's status we need to get it from storage and that will be done only one time and only in case of error happens, so I expect minor but unavoidable performance implications if we want to provide such clarification to error message for pods. Also as I understand all object specific logic related to storage operations must be placed in pkg/registry/<object> packages where it is now.

Agree server must be "client agnostic", will remove the mention of kubectl.